content/blog/2020-03-dplyr-1-0-0-new-verb-features/index.Rmarkdown

```diff
@@ -0,0 +1,231 @@
+---
+title: 'dplyr 1.0.0: new verb features'
+author: Hadley Wickham
+date: '2020-03-20'
+slug: dplyr-1-0-0-new-verb-features
+categories:
+  - package
+tags:
+  - dplyr
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+As we've mentioned, [dplyr 1.0.0](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-is-coming-soon/) is on its way. Today, we've started the official release process by notifying maintainers of packages that have problems with dplyr 1.0.0, and we're planning for a CRAN release on May 1.
+
+In this post, I'll introduce you to the major changes in existing dplyr verbs, along with couple of new verbs. I'll give you updates on `summarise()`, `select()` + `rename()`, `mutate()`, and the join functions.
+
+```{r setup, message = FALSE}
+library(dplyr)
+```
+
+## `summarise()` can produce multiple rows and columns
+
+Two big changes make `summarise()` much more flexible. A single summary expression can now return:
+
+* A vector of any length, creating multiple rows.
+* A data frame, creating multiple columns.
+
+In other words, `summarise()` used to be limited to generating a single value (one row, one column), and now we've lifted that restriction so it can generate a rectangle of multiple rows and columns. This change makes `summarise()` as powerful as the now superseded `do()`, and if you were previously solving this type of problem with list-columns allows you to reduce your usage of `tidyr::unnest()`.
+
+This is a big change to `summarise()` but it should have minimal impact on existing code because it _broadens_ the interface: all existing code will continue to work, and a number of inputs that would have previously errored now work. 
+
+### Quantiles
+
+To demonstrate this new feature we'll start by looking at a summary that was previous hard to compute with `summarise()`. `quantile()` was hard to use because it returns multiple values which used to cause `summarise()` to error. Now it's straightforward:
```

Comment:
```suggestion
To demonstrate this new feature we'll start by looking at a summary that was previously hard to compute with `summarise()`. `quantile()` was hard to use because it returns multiple values which used to cause `summarise()` to error. Now it's straightforward:
```